promdens.py

```diff
@@ -495,29 +557,29 @@ def windowing(self):
-parser = argparse.ArgumentParser(description="Parser for this code", formatter_class=argparse.ArgumentDefaultsHelpFormatter)
-parser.add_argument("-m", "--method", default='pda', type=str,
+parser = argparse.ArgumentParser(description=DESC, formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+parser.add_argument("-m", "--method", default='pda', choices=METHODS,
                     help="Select either Promoted density approach (PDA) to generate initial conditions with excitation times or "
-                         "PDA for windowing (PDAW) to generate weights and convolution parameters. Options: 'pda', 'pdaw'.")
-parser.add_argument("-n", "--nsamples", default=0, type=int,
-                    help="Number of data from the input file considered. 0 takes all initial conditions provided in the input file.")
-parser.add_argument("-np", "--npsamples", default=1000, type=int, help="Number of initial conditions generated with PDA.")
-parser.add_argument("-ns", "--nstates", default=1, type=int, help="Number of excited states considered.")
-parser.add_argument("-ft", "--file_type", default='file', help="Input file type. Options: 'file'.")
-parser.add_argument("-p", "--plot", action="store_true", help="Plot the input data and calculated results. Plots are saved as png images.")
-parser.add_argument("-eu", "--energy_units", default='a.u.', help="Units in which energies are provided. Options: 'a.u.', 'eV', 'nm', 'cm-1'. ")
-parser.add_argument("-tu", "--tdm_units", default='a.u.',
-                    help="Units in which magnitudes of transition dipole moments (|mu_ij|) are provided. Options: 'a.u.', 'debye'.")
-parser.add_argument("-w", "--omega", default=0.1, type=float, help="Frequency of the field in a.u.")
+                         "PDA for windowing (PDAW) to generate weights and convolution parameters.")
+parser.add_argument("-n", "--nsamples", default=0, type=positive_int,
+                    help="Number of data points from the input file considered. By default all initial conditions provided in the input file are taken.")
+parser.add_argument("-np", "--npsamples", default=1000, type=positive_int, help="Number of initial conditions generated with PDA.")
+parser.add_argument("-ns", "--nstates", default=1, type=positive_int, help="Number of excited states considered.")
+parser.add_argument("-ft", "--file_type", choices=FILE_TYPES, default='file', help="Input file type.")
+parser.add_argument("-p", "--plot", action="store_true", help="Plot the input data and calculated results and save them as png images.")
+parser.add_argument("-eu", "--energy_units", choices=ENERGY_UNITS, default='a.u.', help="Units in which excitation energies are provided.")
+parser.add_argument("-tu", "--tdm_units", choices=TDM_UNITS, default='a.u.',
+                    help="Units in which magnitudes of transition dipole moments (|mu_ij|) are provided.")
+parser.add_argument("-w", "--omega", required=True, type=positive_float, help="Frequency of the field in a.u.")
```

Comment:
I removed the default values for omega and fwhm as we discussed yesterday, and added a check that they are positive.